fennel/datasets/datasets.py

```diff
@@ -2770,12 +2796,46 @@ def visitDrop(self, obj) -> DSSchema:
             if field not in val_fields:
                 raise ValueError(
                     f"Field `{field}` is a key or timestamp field in schema of "
-                    f"{obj.name} node input {input_schema.name}. Value fields are: {list(val_fields)}"
+                    f"drop node input {input_schema.name}. Value fields are: {list(val_fields)}"
                 )
         output_schema = obj.dsschema()
         output_schema.name = output_schema_name
         return output_schema
 
+    def visitSelect(self, obj) -> DSSchema:
+        input_schema = copy.deepcopy(self.visit(obj.node))
+        if input_schema.is_terminal:
+            raise ValueError(
+                f"Cannot add node 'Select' after a terminal node in pipeline : `{self.pipeline_name}`."
+            )
+
+        timestamp = input_schema.timestamp
+        key_fields = input_schema.keys
+
+        output_schema_name = f"'[Pipeline:{self.pipeline_name}]->select node'"
+        if obj.columns is None or len(obj.columns) == 0:
+            raise ValueError(
+                f"invalid select - {output_schema_name} must have at least one column to select"
+            )
+        if timestamp not in obj.columns:
+            raise ValueError(
+                f"invalid select - {output_schema_name} timestamp_field : `{timestamp}` must be in columns"
```

Comment:
Super nit but can you please remove the underscore from `timestamp_field` and make it two words?

Reply:
changed